packages/sdk/src/components/expand-record/ExpandRecorder.tsx

```diff
@@ -72,6 +73,15 @@ export const ExpandRecorder = (props: IExpandRecorderProps) => {
     return <></>;
   }
 
+  const onDuplicate = async (tableId: string, recordId: string) => {
+    await duplicateRecords(tableId, recordId, {
+      viewId: viewId || '',
+      anchorId: recordId,
+      position: 'after',
+    });
+    toast({ description: t('expandRecord.duplicateRecord') });
+  };
+
```

Comment:
After a successful copy, the current expand card should be updated with the new record, rather than just closing the expand card.

Reply:
Yes, I found it. I know that `recordsServerData` triggers rendering. When I duplicate record, the length of `recordsServerData` keep same. I don't know how to update RecordsServerData.

Reply:
> Yes, I found it. I know that `recordsServerData` triggers rendering. When I duplicate record, the length of `recordsServerData` keep same. I don't know how to update RecordsServerData.

In fact, recordsServerData will not be updated, recordsServerData is only the initialization data needed for ssr, and will not be used when the page is rendered and the socket is connected, so don't worry about it!

Reply:
Oh, I see. Now I have to force refresh the page for the duplicated record to render.  I have no cules to solve it.

Reply:
Try it.


``` ts
// source
const createdRecords =await this.createRecords(tableId, createRecordsRo);

// aliternative
const createdRecords = await this.prismaService.$tx(
      async () => {
        return await this.createRecords(tableId, createRecordsRo);
      }   
);
```

Reply:
thank you ,it works